pkg/storage/unified/sql/db/dbimpl/db_engine.go

```diff
@@ -69,6 +67,47 @@ func getEngineMySQL(getter confGetter) (*xorm.Engine, error) {
 	return engine, nil
 }
 
+func configureTLS(getter confGetter, config *mysql.Config) error {
+	sslMode := getter.String("ssl_mode")
+
+	if sslMode == "skip-verify" || sslMode == "preferred" {
+		config.Params["tls"] = "preferred"
```

Comment:
It seems that we use a slightly different implemention in grafana [here](https://github.com/grafana/grafana/blob/0854bd801b6cb35645dfbfe044bb031436a176a5/pkg/services/sqlstore/database_config.go#L154)

We set `tls=custom` when `ssl_mode=true` || `ssl_mode=skip-verify`

We also set InsecureSkipVerify in the TLS config.
https://github.com/grafana/grafana/blob/0854bd801b6cb35645dfbfe044bb031436a176a5/pkg/services/sqlstore/tls_mysql.go#L36-L37

Reply:
`custom` is just the [name](https://github.com/grafana/grafana/blob/0854bd801b6cb35645dfbfe044bb031436a176a5/pkg/services/sqlstore/database_config.go#L150) of the config. I used another one to not get conflicts with the multiple providers we have. I added the skip verify part.